### PR TITLE
Feature/create box by dragging left

### PR DESCRIPTION
**Pull request by henriettemoe**
You can now create boxes by dragging to the left and/or upwards.

### CODE REVIEW

File: src/components/Draggable/Draggable.module.scss [hunk 1]
```diff
@@ -18,6 +18,10 @@
   will-change: transform;
   z-index: 1;
 
+  &.preview {
+    transition: none !important;
```
Comment (boyum) [inline src/components/Draggable/Draggable.module.scss:22]: Is the `!important` necessary? If so, why?

File: src/components/Grid/Grid.tsx [hunk 2]
```diff
@@ -159,21 +159,37 @@ export const Grid: React.FC<GridProps> = ({
...
+
         // Get x and y percentage position
-        const x = boxStartPosition % numberOfColumns;
-        const y = Math.floor(boxStartPosition / numberOfColumns);
+        const x = !dragLeft
+          ? boxStartPosition % numberOfColumns
+          : indicatorIndex % numberOfColumns;
+        const y = !dragUp
+          ? Math.floor(boxStartPosition / numberOfColumns)
+          : Math.floor(indicatorIndex / numberOfColumns);
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:176]: Can we flip these?

```suggestion
        const x = dragLeft
          ? indicatorIndex % numberOfColumns
          : boxStartPosition % numberOfColumns;
        const y = dragUp
          ? Math.floor(indicatorIndex / numberOfColumns)
          : Math.floor(boxStartPosition / numberOfColumns);
```

File: src/components/Grid/Grid.tsx [hunk 3]
```diff
@@ -159,21 +159,37 @@ export const Grid: React.FC<GridProps> = ({
...
 
         const xPercentagePosition = (x / numberOfColumns) * 100;
         const yPercentagePosition = (y / numberOfRows) * 100;
 
         // Get height percentage
-        const yEnd = Math.floor(indicatorIndex / numberOfColumns);
+        const yEnd = !dragUp
+          ? Math.floor(indicatorIndex / numberOfColumns)
+          : Math.floor(boxStartPosition / numberOfColumns);
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:184]:
```suggestion
        const yEnd = dragUp
          ? Math.floor(boxStartPosition / numberOfColumns)
          : Math.floor(indicatorIndex / numberOfColumns);
```

File: src/components/Grid/Grid.tsx [hunk 4]
```diff
@@ -159,21 +159,37 @@ export const Grid: React.FC<GridProps> = ({
...
         const yEndPercentagePosition = ((yEnd + 1) / numberOfRows) * 100;
 
         const heightPercentage = yEndPercentagePosition - yPercentagePosition;
 
         // Get width percentage
-        const indicatorValue = indicatorIndex + 1;
+        const indicatorValue = !dragLeft
+          ? indicatorIndex + 1
+          : boxStartPosition + 1;
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:192]:
```suggestion
        const indicatorValue = dragLeft
          ? boxStartPosition + 1
          : indicatorIndex + 1;
```

File: src/components/Draggable/Draggable.module.scss (hunk 1 above)
Comment (henriettemoe) [reply src/components/Draggable/Draggable.module.scss]: Works without, so I'll remove it

**Review by boyum — APPROVED** [review]
✨

**Pull request merged**